app/helpers/container_route_helper/textual_summary.rb

```diff
@@ -28,4 +28,38 @@ def textual_creation_timestamp
   def textual_resource_version
     {:label => "Resource Version", :value => @record.resource_version}
   end
+
+  def textual_ems
+    ems = @record.ext_management_system
+    return nil if ems.nil?
+    label = ui_lookup(:table => "ems_container")
+    h = {:label => label, :image => "vendor-#{ems.image_name}", :value => ems.name}
+    if role_allows(:feature => "ems_container_show")
+      h[:title] = "Show parent #{label} '#{ems.name}'"
+      h[:link] = url_for(:controller => 'ems_container', :action => 'show', :id => ems)
+    end
+    h
+  end
+
+  def textual_container_project
+    project = @record.container_project
+    return nil if project.nil?
+    label = ui_lookup(:table => "container_project")
+    h = {:label => label, :image => "container_project", :value => project.name}
+    if role_allows(:feature => "container_project_show")
+      h[:title] = "Show parent #{label} '#{project.name}'"
+      h[:link] = url_for(:controller => 'container_project', :action => 'show', :id => project)
+    end
+    h
+  end
+
+  def textual_container_service
```

Comment:
This one is special, remove entirely this method from here and re-implement it in `app/helpers/container_summary_helper.rb` (before `textual_container_services`). Code should be:

    def textual_container_service
      textual_single_relationship(
        :container_service,
        :name,
        "container_service",
        "container_service_show",
      )
    end